tests/tests_pytorch/test_cli.py

```diff
@@ -562,6 +577,62 @@ def add_arguments_to_parser(self, parser):
     assert cli.model.layer.out_features == 4
 
 
+def test_adapt_checkpoint_hparams_hook_pop_keys(cleandir):
+    """Test that the adapt_checkpoint_hparams hook is called and modifications are applied."""
+
+    class AdaptHparamsCLI(LightningCLI):
+        def adapt_checkpoint_hparams(self, subcommand: str, checkpoint_hparams: dict) -> dict:
+            """Remove out_dim and hidden_dim for non-fit subcommands."""
+            if subcommand != "fit":
+                checkpoint_hparams.pop("out_dim", None)
+                checkpoint_hparams.pop("hidden_dim", None)
```

Comment:
From a testing perspective there is no difference between `out_dim` and `hidden_dim`. It might be better if one is popped and the other not, so that both cases are tested?